packages/ui-extensions/src/surfaces/admin/api/customer-segmentation-templates/customer-segmentation-templates.ts

```diff
@@ -0,0 +1,58 @@
+import type {StandardApi} from '../standard/standard';
+import type {ExtensionPoint as AnyExtensionPoint} from '../../extension-points';
+
+type ValueOrPromise<T> = T extends PromiseLike<any> ? T : T | Promise<T>;
+
+export enum EnabledCustomerSegmentationFeature {
+  ProductsPurchasedOnTags,
+  AggregateFilters,
+}
+
+export interface SegmentationTemplatesApi<
+  ExtensionPoint extends AnyExtensionPoint,
+> extends StandardApi<ExtensionPoint> {
+  /** Enabled segmentation query language features given to the extension point. */
+  enabledFeatures: EnabledCustomerSegmentationFeature;
```

Comment:
This is required by the extension to do conditional rendering of templates based on the supported QL features